gbpservice/neutron/nsf/configurator/api/v1/controllers/controller.py

```diff
@@ -0,0 +1,181 @@
+import json
+import subprocess
+
+from neutron.common import rpc as n_rpc
+from neutron.agent.common import config
+from oslo_config import cfg
+import oslo_messaging
+from pecan import expose, request
+from pecan import rest
+
+import constants
+
+
+class Controller(rest.RestController):
+
+    """controller class for handling all the curl request"""
+
+    def __init__(self, module_name):
+        self.host = subprocess.check_output('hostname', shell=True).rstrip()
+        self.rpcclient = RPCClient(topic=constants.TOPIC, host=self.host)
+        self.module_name = module_name
+        super(Controller, self).__init__()
+
+    @expose(method='GET', content_type='application/json')
+    def get(self):
+        try:
+            return self._get_notifications()
+        except Exception as e:
+            return json.dumps({'err_msg': e.args})
+
+    @expose(method='POST', content_type='application/json')
+    def post(self, **body):
+        try:
+            body = None
+            if request.is_body_readable:
+                body = request.json_body
+            if self.module_name == "create_network_device_config":
```

Comment:
Can we construct the method name using getattr? It will be single line if we do this.